src/js/utils/architecture.ts

```diff
@@ -0,0 +1,27 @@
+import { getGlobalObject } from '@sentry/utils';
```

Comment:
I liked the refactoring, well done, maybe we just rename the file name? `architecture` is too generic.
`hermes` is past "new" architecture, so it's semantically wrong.

Reply:
Yeah, I know `hermes` is out of the new arch, that why I called it only `architecture`.

What about `runtime`?

Reply:
your call, just wanted to say that maybe we could better name this, no strong opinion tho.

Reply:
Okay, I'll go with `environment`. It seems fitting to me and also would make sense for the future (for example getRNVersion...)